### ISSUE TITLE
ignoreAnnotated does not recognize Compose multipreview annotations

### ISSUE THREAD

**Issue by loganrosen** [detekt/detekt#9562]
## Expected Behavior

When a rule is configured with:

```yaml
ignoreAnnotated:
  - Preview
```

functions annotated with a Compose multipreview annotation should also be ignored when that annotation is itself annotated with `@Preview`.

For example:

```kotlin
@Preview(name = "Light")
@Preview(name = "Dark", uiMode = Configuration.UI_MODE_NIGHT_YES)
annotation class AppPreviews

@AppPreviews
@Composable
private fun ExamplePreview() = Unit
```

This would also cover official Compose multipreview annotations such as `@PreviewLightDark`.

## Current Behavior

`ignoreAnnotated` matches only annotations directly present on the declaration. `UnusedPrivateFunction` therefore still reports `ExamplePreview`, and the annotation name must be added explicitly:

```yaml
ignoreAnnotated:
  - Preview
  - AppPreviews
  - PreviewLightDark
```

Detekt's Compose guidance currently recommends ignoring `Preview` without noting that multipreview annotations require separate entries.

## Context

Compose supports custom multipreview annotations by annotating an annotation class with one or more `@Preview` annotations. Treating these preview functions as ordinary unused private functions creates false positives and baseline entries.

If recursively recognizing meta-annotations is not desirable for `ignoreAnnotated`, documenting the direct-only behavior and updating the Compose guidance for multipreview annotations would still help.

- Detekt Compose guidance: https://detekt.dev/docs/2.0.0-alpha.5/introduction/compose/#unusedprivatefunction-for-compose
- Compose multipreview documentation: https://developer.android.com/develop/ui/compose/tooling/previews#multi-preview

**Comment by cortinico** [comment]
> documenting the direct-only behavior and updating the Compose guidance for multipreview annotations would still help.

I'd suggest we would update the docs as suggested here.

In theory we could resolve "transitive" annotations, but I can image scenarios where users don't want this specific behavior.

In this particular case, I'd rather be explicit with the annotations the user really want to ignore and not try to "be smart".

**Comment by BraisGabin** [comment]
Also, `ignoreAnnotated` works in light mode too. And we can't check the "transitive" annotations in that mode. So I fully agree with @cortinico's proposal to fix this.
